server/src/main/java/io/littlehorse/common/model/metadatacommand/subcommand/DeletePrincipalRequestModel.java

```diff
@@ -0,0 +1,43 @@
+package io.littlehorse.common.model.metadatacommand.subcommand;
+
+import com.google.protobuf.Message;
+import io.littlehorse.common.LHServerConfig;
+import io.littlehorse.common.dao.MetadataProcessorDAO;
+import io.littlehorse.common.model.ServerSubCommand;
+import io.littlehorse.common.model.getable.objectId.PrincipalIdModel;
+import io.littlehorse.common.model.metadatacommand.MetadataSubCommand;
+import io.littlehorse.common.proto.DeletePrincipalRequest;
+import io.littlehorse.sdk.common.exception.LHSerdeError;
+
+public class DeletePrincipalRequestModel extends MetadataSubCommand<DeletePrincipalRequest>
+        implements ServerSubCommand {
+
+    private String id;
+
+    @Override
+    public void initFrom(Message proto) throws LHSerdeError {
+        DeletePrincipalRequest deleteRequest = (DeletePrincipalRequest) proto;
+        this.id = deleteRequest.getId();
+    }
+
+    @Override
+    public DeletePrincipalRequest.Builder toProto() {
+        return DeletePrincipalRequest.newBuilder().setId(id);
+    }
+
+    @Override
+    public Class<DeletePrincipalRequest> getProtoBaseClass() {
+        return DeletePrincipalRequest.class;
+    }
+
+    @Override
+    public boolean hasResponse() {
+        return false;
+    }
+
+    @Override
+    public Message process(MetadataProcessorDAO dao, LHServerConfig config) {
```

Comment:
Can you put some checking here to ensure that the `Principal` isn't the last "admin" left?

Reply:
This will be addressed here in LH-240